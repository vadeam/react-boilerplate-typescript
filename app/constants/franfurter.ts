export const DEFAULT_BASE_CURRENCY = 'EUR'
