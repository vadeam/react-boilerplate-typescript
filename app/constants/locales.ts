export const LOCALE_DE = 'de'
export const LOCALE_EN = 'en'
export const DEFAULT_LOCALE = LOCALE_EN

export const appLocales: Locales[] = [LOCALE_EN, LOCALE_DE]
export const CUSTOM_LOCALE_STORAGE_KEY = 'custom-locale'
