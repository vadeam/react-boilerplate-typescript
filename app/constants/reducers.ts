export const REDUCER_KEY_APP = 'app'
export const REDUCER_KEY_LANGUAGE = 'language'
export const REDUCER_KEY_ROUTER = 'router'
export const REDUCER_KEY_HOME = 'home'
// for testing purposes
export const REDUCER_KEY_TEST = 'test'
