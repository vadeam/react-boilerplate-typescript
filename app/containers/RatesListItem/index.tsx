/**
 * RepoListItem
 *
 * Lists the name and the issue count of a repository
 */

import React from 'react'
import { FormattedNumber } from 'react-intl'

import ListItem from 'components/ListItem'

import { WrapperStyled } from './styled'

export type RateItem = { id: string; currency: string; rate: number }
type RatesListItemProps = { item: RateItem }

export default function RatesListItem({ item: { id, currency, rate } }: RatesListItemProps) {
  const content = (
    <WrapperStyled className="currency-rate">
      {currency}:&nbsp;&nbsp;
      <FormattedNumber value={rate} maximumFractionDigits={4} />
    </WrapperStyled>
  )

  return <ListItem key={`rates-list-item-${id}`} item={content} />
}
