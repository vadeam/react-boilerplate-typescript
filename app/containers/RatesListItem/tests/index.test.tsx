/**
 * Test the repo list item
 */

import { render, getByText } from '@testing-library/react'
import React from 'react'
import { IntlProvider } from 'react-intl'

import { LOCALE_EN } from 'constants/locales'

import RatesListItem, { RateItem } from '../index'

const renderComponent = (item: RateItem) =>
  render(
    <IntlProvider locale={LOCALE_EN}>
      <RatesListItem item={item} />
    </IntlProvider>,
  )

describe('<RepoListItem />', () => {
  let item: RateItem

  beforeEach(() => {
    item = { id: 'USD', currency: 'USD', rate: 1.2345 }
  })

  it('should render a ListItem', () => {
    const { container } = renderComponent(item)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render usernames that are not the current one', () => {
    const { container } = renderComponent(item)
    expect(getByText(container, (content) => content.startsWith(item.currency))).not.toBeNull()
  })

  it('should render the currency name and rate', () => {
    const { container } = renderComponent(item)

    const element = container.getElementsByClassName('currency-rate')

    expect(element.length).toEqual(1)
    expect(element.item(0)).not.toBeNull()
    // @ts-ignore
    expect(element.item(0).innerHTML).toEqual(`${item.currency}:&nbsp;&nbsp;${item.rate}`)
  })

  it('should render the li', () => {
    const { container } = renderComponent(item)
    expect(container.querySelector('li')).not.toBeNull()
  })
})
