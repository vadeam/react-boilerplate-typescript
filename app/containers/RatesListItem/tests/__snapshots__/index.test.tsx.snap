// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoListItem /> should render a ListItem 1`] = `
<li
  class="sc-gsTEea bNKyBL"
>
  <div
    class="sc-bdfBQB hNUSPl"
  >
    <div
      class="sc-dlfnuX eXHPyD currency-rate"
    >
      USD
      :  
      1.2345
    </div>
  </div>
</li>
`;
